pkg/app/mockVersionResolver.go

```
@@ -0,0 +1,64 @@
+package app
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+type MockVersionResolver struct {
```

Slight nit, I like to build mock providers that store data instead of explicit `ReturnsErr` options. This makes it a little more natural to use (since If you operate on what you think it'd return, it's easier to determine if it should return an error or not).

Here's an example: https://github.com/getoutreach/argocd/blob/master/maestro-dv-syncer/internal/argocd/argocdtest/argocdtest.go

---

Hmm, I spent a few trying to figure out what to store as data but the only data that gets returned is the version (string) and error if its unable to get the version from that version resolver. Maybe we can chat about this more in the meeting I added for Monday next week

---

```
type VersionResolver interface {
	GetLatestVersion(context.Context, logrus.FieldLogger, *App) (string, error)
}
```